### PR TITLE
Unclean shutdown on SIGINT / SIGTERM

### PR DESCRIPTION
**Pull request by melekes**
Refs #3238

* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [ ] ~~Wrote tests~~
* [x] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: docs/app-dev/abci-cli.md [hunk 1]
```diff
@@ -238,12 +240,14 @@ func cmdCounter(cmd *cobra.Command, args []string) error {
         return err
     }
 
-    // Wait forever
-    cmn.TrapSignal(func() {
+	  // Stop upon receiving SIGTERM or CTRL-C.
```
Comment (thanethomson) [inline docs/app-dev/abci-cli.md]:
```suggestion
    // Stop upon receiving SIGTERM or CTRL-C.
```

There's a mixture of tabs and spaces here, throws the formatting off.

File: cmd/tendermint/commands/lite.go [hunk 2]
```diff
@@ -59,6 +59,11 @@ func ensureAddrHasSchemeOrDefaultToTCP(addr string) (string, error) {
 }
 
 func runProxy(cmd *cobra.Command, args []string) error {
+	// Stop upon receiving SIGTERM or CTRL-C.
+	cmn.TrapSignal(logger, func() {
+		// TODO: close up shop
```
Comment (alexanderbez) [inline cmd/tendermint/commands/lite.go]: Is there anything of significant importance to do here?

File: rpc/lib/test/main.go [hunk 3]
```diff
@@ -24,17 +24,19 @@ type Result struct {
 }
 
 func main() {
-	mux := http.NewServeMux()
-	cdc := amino.NewCodec()
-	logger := log.NewTMLogger(log.NewSyncWriter(os.Stdout))
+	var (
```
Comment (alexanderbez) [inline rpc/lib/test/main.go]: ❤️

File: cmd/tendermint/commands/lite.go (hunk 2 above)
Comment (melekes) [reply cmd/tendermint/commands/lite.go]: stop the server? created https://github.com/tendermint/tendermint/issues/3318

File: docs/app-dev/abci-cli.md [hunk 4]
```diff
@@ -89,12 +89,14 @@ func cmdKVStore(cmd *cobra.Command, args []string) error {
...
```
Comment (Liamsi) [inline docs/app-dev/abci-cli.md]:
```suggestion
    // Stop upon receiving SIGTERM or CTRL-C.
```

**Pull request merged**